Change default smart connection to ssh on macOS and remove paramiko from requirements.txt (#54738)
ansible/ansible: docs/docsite/rst/installation_guide/intro_installation.rst

Latest Releases via Pip
+++++++++++++++++++++++

Ansible can be installed via ``pip``, the Python package manager.  If ``pip`` isn't already available in
your version of Python, you can get ``pip`` by::

    $ curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
    $ sudo python get-pip.py

Then install Ansible with [1]_::

    $ sudo pip install ansible

Or if you are looking for the latest development version::

    $ sudo pip install git+https://github.com/ansible/ansible.git@devel

If you are installing on macOS Mavericks, you may encounter some noise from your compiler. A workaround is to do the following::

    $ sudo CFLAGS=-Qunused-arguments CPPFLAGS=-Qunused-arguments pip install ansible

If you need to use the ``paramiko`` connection plugin, install the required module[2]_::

    $ sudo pip install paramiko

Ansible can also be installed in the user directory by using ``--user``::

>     $ python get-pip.py --user

Reviewer: since you're in the neighborhood- should we make this the recommended method instead of global (since pip does the same)?

Author: I did remove the line about "just install in globally" that seemed a bit glib. I'll just reverse this suggestion, ``--user`` by default but ``sudo`` if you want it globally with big warning.

Latest Releases via Pip
+++++++++++++++++++++++

Ansible can be installed via ``pip``, the Python package manager.  If ``pip`` isn't already available on your system of Python, run the following commands to install it::

    $ curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
    $ python get-pip.py --user

Then install Ansible [1]_::

    $ pip install --user ansible

Or if you are looking for the latest development version::

    $ pip install --user git+https://github.com/ansible/ansible.git@devel

If you are installing on macOS Mavericks (10.9), you may encounter some noise from your compiler. A workaround is to do the following::

    $ CFLAGS=-Qunused-arguments CPPFLAGS=-Qunused-arguments pip install --user ansible

> If you need to use the ``paramiko`` connection plugin, install the required module [2]_::

Reviewer 2: How would a user know if s/he needs to use the paramiko connection plugin?

Author: That's a good question. The only reason I put this in here is because `paramiko` used to get installed with Ansible by default. As of 2.8, it will not be. Most users on macOS have been using `paramiko` as the transport but just did not know. Or if some is used to using `paramiko` as the transport, they will be surprised that it is not longer automatically installed since it has been for a very long time (if not always).

One case where they would need to install `paramiko` is if they are using password authentication with `ssh` connection plugin and are unable to install `sshpass` (it's _slightly_ difficult to install on macOS). `sshpass` is required to use passwords with `ssh`, but isn't required for key-based authentication.

That's a lot of caveats to put there, so I'm not sure how to say all that succinctly. :smile:

Author: I guess I can add some more info on what you would want that in the footnote rather than cluttering up the directions.

    $ pip install --user paramiko

Ansble can also be installed inside a new or existing ``virtualenv``::

    $ python -m virtualenv ansible  # Create a virtualenv if one does not already exist
    $ source ansible/bin/activate   # Activate the virtual environment
    $ pip install ansible

If you wish to install Ansible globally, run the following commands::

    $ sudo python get-pip.py
    $ sudo pip install ansible

.. note::

    Running ``pip`` with ``sudo`` will make global changes to the system. Since ``pip`` does not coordinate with system package managers, it could make changes to you system that leave it in an inconsistent on non-functioning state. This is particularly true for macOS. Installing with ``--user`` is recommended unless you understand fully the implications of modifying global files on the system.

.. note::

    Older versions of ``pip`` default to http://pypi.python.org/simple, which no longer works.
    Please make sure you have an updated ``pip`` (version 10 or greater) installed before installing Ansible.

>     Refer `here <https://pip.pypa.io/en/stable/installing/#installation>`_ about installing latest ``pip``.

Reviewer 2: This last sentence is a bit confusing - I'm guessing on the intended meaning here, but I'd change this to:

```suggestion
    If you have an older version of ``pip`` installed, you can upgrade by following `pip's upgrade instructions <https://pip.pypa.io/en/stable/installing/#upgrading-pip>`_ .
```

Latest Releases via Pip
+++++++++++++++++++++++

Ansible can be installed via ``pip``, the Python package manager.  If ``pip`` isn't already available in
your version of Python, you can get ``pip`` by::

>    $ sudo easy_install pip

Reviewer 3: Since you're updating docs you should probably update the pip install instructions:

https://pip.pypa.io/en/stable/installing/#installing-with-get-pip-py

Running From Source
+++++++++++++++++++

> Ansible is easy to run from a checkout. ``root`` permissions are not required

Reviewer 2: ```suggestion
Ansible is easy to run from a checkout. You do not need ``root`` permissions
```

to use it and there is no software to actually install. No daemons
or database setup are required. Because of this, many users in our community use the
development version of Ansible all of the time so they can take advantage of new features
when they are implemented and easily contribute to the project. Because there is
nothing to install, following the development version is significantly easier than most
open source projects.

.. note::

>    If you are intending to use Tower as the Control Node, do not use a source install. Please use OS package manager (like ``apt`` or ``yum``) or ``pip`` to install a stable version.

Reviewer 2: ```suggestion
   If you want to use Ansible Tower as the Control Node, do not use a source install of Ansible. Please use OS package manager (like ``apt`` or ``yum``) or ``pip`` to install a stable version.
```

    $ source ./hacking/env-setup -q

If you don't have ``pip`` installed in your version of Python, install it::

    $ curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
    $ python get-pip.py --user

Ansible also uses the following Python modules that need to be installed [1]_:

.. code-block:: bash

    $ pip install --user -r ./requirements.txt

To update ansible checkouts, use pull-with-rebase so any local changes are replayed.

.. code-block:: bash

    $ git pull --rebase

.. note::

>     When updating Ansible checkouts that are v2.2 and older, be sure to not

Reviewer 2: Can we remove this note entirely, since we no longer support anything older than 2.5?
